jme3-plugins/src/main/java/com/jme3/material/plugin/export/material/J3MExporter.java

```diff
@@ -50,13 +50,14 @@ public void save(Savable object, OutputStream f) throws IOException {
             throw new IllegalArgumentException("J3MExporter can only save com.jme3.material.Material class");
         }
 
-        OutputStreamWriter out = new OutputStreamWriter(f, Charset.forName("UTF-8"));
+        try (OutputStreamWriter out = new OutputStreamWriter(f, StandardCharsets.UTF_8);
+                BufferedWriter writer = new BufferedWriter(out)) {
```

Comment:
I am not sure about this.
The method might be called to write on a stream that doesn't need (or have already) buffering cache.
I think this should be done like in the other PR, where the Buffered*Steam is used on the method that writes on files.

Reply:
That is true.  The method signature changes are really good, though.  Requiring an OutputStreamWriter was kind of silly.

Reply:
Yeah, I also believe that is how Java is generally wanting us to work. If you give input/output stream as param, you want to control how/when/where. The BinaryImporter does violate this where as the BinaryExporter doesn't.

This was a bit controversial change in that sense. I just thought that the worst thing that can happen is double buffering. And BufferedWriter differs from BufferedStream a bit, I think it also adds cache to the characters. And you can't do this anymore from outside then. Only give a stream that buffers.

I think all the usages of this are outside the engine. But I can check that and leave just the method signature changes.

Reply:
> I just thought that the worst thing that can happen is double buffering.

It can reduce performances in common use cases and add extra delays that are unexpected.

>  And BufferedWriter differs from BufferedStream a bit, I think it also adds cache to the characters.

But if this is piped to a BufferedOutputStream, everything will be cached anyway.

> I think all the usages of this are outside the engine. But I can check that and leave just the method signature changes.

The signature changes are ok, and i think  a BufferedOutputStream should be used here

https://github.com/jMonkeyEngine/jmonkeyengine/blob/d25ab90bb99c95850d059791026d0a2428d09c3d/jme3-plugins/src/main/java/com/jme3/material/plugin/export/material/J3MExporter.java#L65-L67

Similarly to what you did for the importer, since i believe it is reasonable to assume that this code will always benefit from this change.

Reply:
Yep, changed those. Also did the trick to the MaterialDefinitionExporter.